includes/data-stores/class-wcs-related-order-data-store.php

```diff
@@ -0,0 +1,201 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Related order data store for orders.
+ *
+ * Importantly, this class uses WC_Data API methods, like WC_Data::add_meta_data() and WC_Data::get_meta(), to manage the
+ * relationships instead of add_post_meta() or get_post_meta(). This ensures that the relationship is stored, regardless
+ * of the order data store being used.
+ *
+ * @version 2.0.0
+ */
+class WCS_Related_Order_Data_Store extends WCS_Related_Order_Store {
+
+	/**
+	 * Meta keys used to link an order with a subscription for each type of relationship.
+	 *
+	 * @since 2.0.0
+	 * @var array $meta_keys Relationship => Meta key
+	 */
+	private $meta_keys;
+
+	/**
+	 * Constructor: sets meta keys used for storing each order relation.
+	 */
+	public function __construct() {
+		foreach ( $this->get_relation_types() as $relation_type ) {
+			$this->meta_keys[ $relation_type ] = sprintf( '_subscription_%s', $relation_type );
+		}
+	}
+
+	/**
+	 * Find orders related to a given subscription in a given way.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param WC_Order $subscription  The ID of the subscription for which calling code wants the related orders.
+	 * @param string   $relation_type The relationship between the subscription and the orders. Must be 'renewal', 'switch' or 'resubscribe.
+	 *
+	 * @return array
+	 */
+	public function get_related_order_ids( WC_Order $subscription, $relation_type ) {
+		$order_relation_query_var_handler = function( $query, $query_vars ) use ( $subscription, $relation_type ) {
+			if ( ! empty( $query_vars['_related_orders_by_relation_type'] ) ) {
+				$query['meta_query'][] = array(
+					'key'     => $this->get_meta_key( $relation_type ),
+					'compare' => '=',
+					'value'   => $subscription->get_id(), // We can't rely on get_id() being available here, because we only require a WC_Order, not a WC_Subscription, and WC_Order does not have get_id() available with WC < 3.0
```

Comment:
Looks like WCS requires 4.6 & WCPay requires at least 5.2. So this comment can be removed.

```suggestion
					'value'   => $subscription->get_id(),
```